carma_nav2_behavior_tree/test/plugins/action/test_compute_route_action.cpp

```diff
@@ -0,0 +1,250 @@
+    } else {
+      result->path.poses[0].pose.position.x = 0.0;
+    }
+    goal_handle->succeed(result);
+  }
+};
+
+class ComputeRouteActionTestFixture : public ::testing::Test
+{
+public:
+  static void SetUpTestCase()
+  {
+    node_ = std::make_shared<rclcpp::Node>("compute_route_action_test_fixture");
+    factory_ = std::make_shared<BT::BehaviorTreeFactory>();
+
+    config_ = new BT::NodeConfiguration();
+
+    // Create the blackboard that will be shared by all of the nodes in the tree
+    config_->blackboard = BT::Blackboard::create();
+    // Put items on the blackboard
+    config_->blackboard->set("node", node_);
+    config_->blackboard->set<std::chrono::milliseconds>(
+      "server_timeout", std::chrono::milliseconds(20));
+    config_->blackboard->set<std::chrono::milliseconds>(
+      "bt_loop_duration", std::chrono::milliseconds(10));
+    config_->blackboard->set<std::chrono::milliseconds>(
+      "wait_for_service_timeout", std::chrono::milliseconds(1000));
+    config_->blackboard->set("initial_pose_received", false);
+
+    BT::NodeBuilder builder = [](const std::string & name, const BT::NodeConfiguration & config) {
+      return std::make_unique<carma_nav2_behavior_tree::ComputeRouteAction>(
+        name, "compute_route", config);
+    };
+
+    factory_->registerBuilder<carma_nav2_behavior_tree::ComputeRouteAction>(
+      "ComputeRoute", builder);
+  }
+
+  static void TearDownTestCase()
+  {
+    delete config_;
+    config_ = nullptr;
+    node_.reset();
+    action_server_.reset();
+    factory_.reset();
+  }
+
+  void TearDown() override { tree_.reset(); }
+
+  static std::shared_ptr<ComputeRouteActionServer> action_server_;
+
+protected:
+  static rclcpp::Node::SharedPtr node_;
+  static BT::NodeConfiguration * config_;
+  static std::shared_ptr<BT::BehaviorTreeFactory> factory_;
+  static std::shared_ptr<BT::Tree> tree_;
+};
+
+rclcpp::Node::SharedPtr ComputeRouteActionTestFixture::node_ = nullptr;
+std::shared_ptr<ComputeRouteActionServer> ComputeRouteActionTestFixture::action_server_ = nullptr;
+BT::NodeConfiguration * ComputeRouteActionTestFixture::config_ = nullptr;
+std::shared_ptr<BT::BehaviorTreeFactory> ComputeRouteActionTestFixture::factory_ = nullptr;
+std::shared_ptr<BT::Tree> ComputeRouteActionTestFixture::tree_ = nullptr;
+
+TEST_F(ComputeRouteActionTestFixture, test_tick)
+{
+  // create tree
+  std::string xml_txt =
+    R"(
+        <root BTCPP_format="4">
+          <BehaviorTree ID="MainTree">
+              <ComputeRoute goal="{goal}" path="{path}"/>
+          </BehaviorTree>
+        </root>)";
+
+  tree_ = std::make_shared<BT::Tree>(factory_->createTreeFromText(xml_txt, config_->blackboard));
+
+  // create new goal and set it on blackboard
+  geometry_msgs::msg::PoseStamped goal;
+  goal.header.stamp = node_->now();
+  goal.pose.position.x = 1.0;
+  config_->blackboard->set("goal", goal);
+
+  // tick until node succeeds
+  while (tree_->rootNode()->status() != BT::NodeStatus::SUCCESS) {
```

Comment:
This looks like it could easily get stuck in an infinite loop. Can we add a failure condition?